web/src/sidebar_ui.ts

```diff
@@ -94,6 +95,8 @@ export function initialize() {
                 return;
             }
 
+            assert(e.target instanceof Element);
```

Comment:
so `e.target` here is type [`EventTarget`](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget).

> [Element](https://developer.mozilla.org/en-US/docs/Web/API/Element), and its children, as well as [Document](https://developer.mozilla.org/en-US/docs/Web/API/Document) and [Window](https://developer.mozilla.org/en-US/docs/Web/API/Window), are the most common event targets, but other objects can be event targets, too. For example [IDBRequest](https://developer.mozilla.org/en-US/docs/Web/API/IDBRequest), [AudioNode](https://developer.mozilla.org/en-US/docs/Web/API/AudioNode), and [AudioContext](https://developer.mozilla.org/en-US/docs/Web/API/AudioContext) are also event targets.

It seems like we're assuming it's of type `Element` for the rest of this code to work, but maybe we actually just want to return early if it's not an `Element`? I don't understand much yet about what other kinds of targets can end up here and what it means to have those.

Reply:
This handler is registered on any click on the app, so correct would be to return early if it's not of type `Element`.

Reply:
Cool, updated to an early return and pushed that change.